baseplate/events/queue.py

```diff
@@ -74,19 +89,34 @@ def set_field(self, key, value, obfuscate=False):
         if value is None or value == "":
             return
 
-        if not obfuscate:
-            self.payload[key] = value
-        else:
-            obfuscated_payload = self.payload.setdefault("obfuscated_data", {})
-            obfuscated_payload[key] = value
+        if obfuscate:
+            kind = FieldKind.OBFUSCATED
+            warn_deprecated("Passing obfuscate to set_field is deprecated in"
+                            " favor of passing a FieldKind value as kind.")
+
+        self.payload[key] = value
+        self.payload_types[key] = kind
 
     def serialize(self):
+        payload = {
```

Comment:
would it be possible to only add these keys if there's a field that needs to be in them? i think data generally prefers minimalism here.